packages/@sanity/cli/test/shared/globalSetup.ts

```diff
@@ -0,0 +1,187 @@
+
+      const flags = version === 'v2' ? ['--legacy-peer-deps'] : []
+      await exec(npmPath, ['install', '--no-package-lock', ...flags], {cwd: destinationPath})
+
+      // We'll want to test the actual integration with the `sanity` module,
+      // instead of the version that is available on npm
+      if (version === 'v3') {
+        await exec(nodePath, [SYMLINK_SCRIPT, destinationPath], {cwd: destinationPath})
+      }
+    })
+  )
+}
+
+async function prepareCliAuth(configPath: string) {
+  if (!cliUserToken) {
+    throw new Error('SANITY_CI_CLI_AUTH_TOKEN not set')
+  }
+
+  const client = new SanityClient({
+    projectId: cliProjectId,
+    apiVersion: '1',
+    useCdn: false,
+    token: cliUserToken,
+  })
+  const user = await client.users.getById('me')
+  if (!user || !user.id) {
+    throw new Error('CLI auth token did not return a valid user')
+  }
+
+  // Store the config file in a different directory than the default, in order
+  // to be easier to test locally without having the local user conflict
+  const cs = new Configstore('sanity', {}, {globalConfigPath: true, configPath})
+  cs.set('authToken', cliUserToken)
+}
+
+async function prepareDatasets() {
+  const client = new SanityClient({
+    projectId: cliProjectId,
+    apiVersion: '2022-06-03',
+    useCdn: false,
+    token: cliUserToken,
+  })
+
+  for (const version of studioVersions) {
+    const args = getTestRunArgs(version)
+    const datasets = [args.documentsDataset, args.graphqlDataset, args.aclDataset]
+
+    await Promise.all(
+      datasets.map((ds) =>
+        client.datasets.create(ds, {aclMode: 'public'}).catch((err) => {
+          err.message = `Failed to create dataset "${ds}":\n${err.message}`
+          throw err
+        })
+      )
+    )
+  }
+}
+
+async function installAndVerifyPackedCli({
+  packedFilePath,
+}: {
+  packedFilePath: string
+}): Promise<string> {
+  // Install the packed tarball into the folder
+  const [cliManifest] = await Promise.all([
+    await readFile(path.join(__dirname, '..', '..', 'package.json'), 'utf8'),
+    await exec(npmPath, ['install', packedFilePath], {cwd: cliInstallPath}),
+  ])
+
+  // Ensure the referenced binary exists
+  const version = execFileSync(cliBinPath, ['--version'], {encoding: 'utf8'}).trim()
+
+  // Ensure the version matches the one we just installed, eg is the correct build
+  const correctVersion = JSON.parse(cliManifest).version
+
+  if (!version.includes(correctVersion)) {
+    throw new Error(`CLI version should include ${correctVersion} - got ${version}`)
+  }
+
+  return cliBinPath
+}
+
+async function prepareCliInstall(): Promise<void> {
+  // Prepare a folder for the CLI to be installed into
+  await mkdir(cliInstallPath, {recursive: true})
+  spawnSync(npmPath, ['init', '-y'], {cwd: cliInstallPath})
+}
+
+async function packCli(): Promise<string> {
+  // Run `npm pack` so we can create a fully isolated install, replicating what a user would get
```

Comment:
I think this is a good idea – since this means we test what will actually be installed in `node_modules` out there.